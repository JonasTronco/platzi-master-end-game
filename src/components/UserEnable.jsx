import WarningIcon from '../assets/icons/warning.svg';
import '../assets/styles/components/UserEnable.scss';

function UserEnable() {
  return (
    <div className="user-enable">
      <WarningIcon className="user-enable__icon" />
      <h2 className="user-enable__text">Are you sure you want to enable this user?</h2>
      <div className="user-enable__buttons">
        <button className="btn--positive" type="submit">Confirm</button>
        <button className="btn--negative" type="submit">Cancel</button>
      </div>
    </div>
  );
}

export default UserEnable;
