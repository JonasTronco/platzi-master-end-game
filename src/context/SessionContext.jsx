import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
