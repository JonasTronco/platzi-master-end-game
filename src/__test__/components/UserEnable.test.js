import UserEnable from '../../components/UserEnable';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('UserEnable Component', () => {
  test('Should render Login component', () => {
    const component = create(
      <SessionProvider>
        <UserEnable />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
