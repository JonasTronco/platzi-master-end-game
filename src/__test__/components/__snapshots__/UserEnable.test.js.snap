// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEnable Component Should render UserEnable component 1`] = `
<div
  className="user-enable"
>
  <div
    className="user-enable__icon"
  />
  <h2
    className="user-enable__text"
  >
    Are you sure you want to enable this user?
  </h2>
  <div
    className="user-enable__buttons"
  >
    <button
      className="btn--positive"
      type="submit"
    >
      Confirm
    </button>
    <button
      className="btn--negative"
      type="submit"
    >
      Cancel
    </button>
  </div>
</div>
`;
